src/utils.rs

```diff
@@ -93,22 +102,80 @@ impl<'a> InstructionTable<'a> {
         Ok(TypesTable { _table: result })
     }
 
-    pub fn build_executable_binary(&self) -> Result<Vec<u8>> {
-        const MAX_STEP_COUNT: i32 = 1000;
+    fn classify(operation: &wasmparser::Operator) -> InstrumentationTypes {
+        use wasmparser::Operator::*;
+        match operation {
+            LocalSet { local_index } | LocalTee { local_index } => {
+                InstrumentationTypes::LocalSet(*local_index)
+            }
+            GlobalSet { global_index } => InstrumentationTypes::GlobalSet(*global_index),
+            I32Const { .. } => InstrumentationTypes::I32Const,
+            // Binary i32 ops: pop 2, push 1
+            I32Add | I32Sub | I32Mul | I32DivS | I32DivU | I32RemS | I32RemU | I32And | I32Or
+            | I32Xor | I32Shl | I32ShrS | I32ShrU | I32Rotl | I32Rotr | I32Eq | I32Ne | I32LtS
+            | I32LtU | I32GtS | I32GtU | I32LeS | I32LeU | I32GeS | I32GeU => {
```

Comment:
Why are we explicitly matching on these? We should be able to infer from the types table, right?